core/com.b2international.snowowl.api.rest/src/com/b2international/snowowl/api/rest/SnowOwlApiConfig.java

```diff
@@ -49,108 +65,99 @@
 import com.fasterxml.jackson.databind.SerializationFeature;
 import com.fasterxml.jackson.databind.util.ISO8601DateFormat;
 import com.fasterxml.jackson.datatype.guava.GuavaModule;
-import com.mangofactory.swagger.configuration.SpringSwaggerConfig;
-import com.mangofactory.swagger.models.alternates.AlternateTypeRule;
-import com.mangofactory.swagger.paths.RelativeSwaggerPathProvider;
-import com.mangofactory.swagger.plugin.EnableSwagger;
-import com.mangofactory.swagger.plugin.SwaggerSpringMvcPlugin;
-import com.wordnik.swagger.model.ApiInfo;
+import com.google.common.base.Charsets;
+
+import springfox.documentation.schema.AlternateTypeRule;
+import springfox.documentation.service.ApiInfo;
+import springfox.documentation.service.Contact;
+import springfox.documentation.spi.DocumentationType;
+import springfox.documentation.spring.web.plugins.Docket;
+import springfox.documentation.swagger2.annotations.EnableSwagger2;
 
 /**
- * The Spring configuration class for Snow Owl's internal REST services module.
- *
- * @since 1.0
+ * @since 7.0
  */
-@Configuration
-@EnableSwagger
 @EnableWebMvc
-public class ServicesConfiguration extends WebMvcConfigurerAdapter {
-
-	private SpringSwaggerConfig springSwaggerConfig;
-	private ServletContext servletContext;
+@EnableSwagger2
+@Configuration
+@ComponentScan("com.b2international.snowowl.api.rest")
+@Import({ SnowOwlSecurityConfig.class })
+@PropertySource(value="classpath:com/b2international/snowowl/api/rest/service_configuration.properties")
+public class SnowOwlApiConfig extends WebMvcConfigurerAdapter {
 
+	@Value("${api.version}")
 	private String apiVersion;
-
+	
+	@Value("${api.title}")
 	private String apiTitle;
+	
+	@Value("${api.description}")
 	private String apiDescription;
+	
+	@Value("${api.termsOfServiceUrl}")
 	private String apiTermsOfServiceUrl;
+	
+	@Value("${api.contact}")
 	private String apiContact;
+	
+	@Value("${api.license}")
 	private String apiLicense;
+	
+	@Value("${api.licenseUrl}")
 	private String apiLicenseUrl;
 	
-	@Autowired
-	public void setServletContext(final ServletContext servletContext) {
-		this.servletContext = servletContext;
-	}
-
-	@Autowired
-	public void setSpringSwaggerConfig(final SpringSwaggerConfig springSwaggerConfig) {
-		this.springSwaggerConfig = springSwaggerConfig;
-	}
-
-	@Autowired
-	@Value("${api.version}")
-	public void setApiVersion(final String apiVersion) {
-		this.apiVersion = apiVersion;
+	@Bean
+	public AuthenticationProvider authenticationProvider() {
+		return new SnowOwlAuthenticationProvider();
 	}
-
-	@Autowired
-	@Value("${api.title}")
-	public void setApiTitle(final String apiTitle) {
-		this.apiTitle = apiTitle;
+	
+	@Bean
+	public IdentityProvider identityProvider() {
+		return com.b2international.snowowl.core.ApplicationContext.getInstance().getServiceChecked(IdentityProvider.class);
```

Comment:
This can be added as an import statement, we're not referencing (or no longer referencing) Spring's class with the same name.